### ISSUE TITLE
Homepage has a broken left nav

### ISSUE THREAD

**Issue by jaredwilli** [GoogleChrome/devtools-docs#216]
These two pages do not match. The first link the left nav does not show, so it appears to be broken.

https://developer.chrome.com/devtools
https://developer.chrome.com/devtools/index

cc @Meggin

**Comment by Garbee** [comment]
Also the note class is causing the large gap of whitespace at the top. Since the class has width=85% it is causing that to show up below the navigation.

**Comment by Meggin** [comment]
Very weird. Native client index page is having some troubles too-- +jmedley
who's investigating. I suspect the two are related.

Meggin

**Comment by Meggin** [comment]
I haven't found a solution, but I have learned enough to redefine the
problem. I'll keep investigating, but I'm sharing in case it makes someone
have an epiphany.

On the pages where navigation is missing, it's not actually missing. The
height of the nav element and several of its children is set to 0, making
it not visible.

Joe Medley | Technical Writer | jmedley@google.com | 816-678-7195

**Comment by kdzwinel** [comment]
https://developer.chrome.com/devtools
![screen shot 2015-01-12 at 21 36 40](https://cloud.githubusercontent.com/assets/985504/5710531/5424e7e4-9aa3-11e4-936c-fd4a20833213.png)

https://developer.chrome.com/devtools/index
![screen shot 2015-01-12 at 21 36 50](https://cloud.githubusercontent.com/assets/985504/5710542/5f1a0e86-9aa3-11e4-9bea-8c2cd4e9b238.png)

Note the two missing "show" classes in the first one. It seems that these are set by the back-end.

**Comment by Meggin** [comment]
Refining the problem, Part II

The article template (
https://code.google.com/p/chromium/codesearch#chromium/src/chrome/common/extensions/docs/templates/private/article.html&q=removeToc&sq=package:chromium&type=cs)
contains this little bit of code:


    <nav class="inline-toc no-permalink">      {{^removeToc}}
$(table_of_contents)      {{/removeToc}}    </nav>


It has the right name. It's in the right place. The problem is I can't
figure out how to change its behavior. Is that a preview server limitation?


Joe Medley | Technical Writer | jmedley@google.com | 816-678-7195

**Comment by kdzwinel** [comment]
In my opinion `{{^removeToc}}` is not responsible. HTML markup for the left menu *exists*, it's just not visible. Execute this line in the DevTools console and menu will appear:
`$('.inline-site-toc > section > ol > li').classList.add('show')`

Looking at the templates, this is the place where it's decided if 'show' class should be added:
https://code.google.com/p/chromium/codesearch#chromium/src/chrome/common/extensions/docs/templates/private/relatedpages_items.html&sq=package:chromium&type=cs&l=4
Going deeper, we get here:
https://code.google.com/p/chromium/codesearch#chromium/src/chrome/common/extensions/docs/templates/private/show_selected_and_related.html&sq=package:chromium&type=cs&l=5
So it seems that `item:l0` misses flag `child_selected`. From this, I understand that menu is hidden because nothing in the menu is selected. On /devtools/index page 'Overview' is selected. Why 'Overview' is selected? There must be some connection between 'Overview' and '/devtools/index'. That's what I found:

https://code.google.com/p/chromium/codesearch#chromium/src/chrome/common/extensions/docs/templates/json/chrome_sidenav.json&q=/devtools/index&sq=package:chromium&type=cs&l=11

If I had to guess, the solution is to change `"href": "/devtools/index"` to `"href": "/devtools"`. This will probably break the /devtools/index page, but that page is not searchable (AFAIK).

**Comment by Meggin** [comment]
Konrad I think your solution might work. A nice and simple change to the
href in the json. I'll try it today. The fix will be in chromium not
github. I'll update this ticket with the cl. I'm in meetings in the morning
but still hope to get to it before lunch.

**Comment by Meggin** [comment]
Hey, guys.

Konrads quick suggestion didn't work :(. So I went with redirects, which isn't elegant, but it should fix the problem: https://codereview.chromium.org/849303003/

I've created a bug too for a better solution.

**Comment by paulirish** [comment]
@Meggin while merged, the change isn't deployed yet.

**Comment by Meggin** [comment]
Yep. We are having a problem with the server taking on changes. I'll
follow-up once the changes are live.

**Comment by Meggin** [comment]
Changes are live, and unfortunately, redirects aren't working for DevTools.
The fix worked for extensions only. I will investigate adding the new
navigation to the specific article template for DevTools. This will work,
but it takes a little bit more time to implement.

I'll target this week, but it might be towards the end of the week. Is this
OK?

Meggin

**Comment by Garbee** [comment]
Redirects seem to be working for me. docs/elements is forwarding correctly to dom-and-styles. (However this could be because the instances I'm hitting are not updated yet. We have noticed before updates can take some time to propagate to all containers.)

**Comment by Meggin** [comment]
I updated the redirects for the server as I was trying to get a redirect to
work at the top-level url. But actually at the lower level might work-- I'm
going to try a quick fix ("": "index"). It might break stuff, and I'll
revert quickly.

Meggin

**Comment by Meggin** [comment]
Pushed the update and waiting to test live (might take a little bit).

Meggin

**Comment by paulirish** [comment]
this live?

**Comment by Meggin** [comment]
It should be, and if you navigate to devtools, the toc is there (which was
the original fix needed). Annoyingly though, devtools/index isn't working.
It's OK for native-client and extensions now, but not for devtools.

I'll look into it.

**Comment by Meggin** [comment]
It's annoying that we can't test these fixes for the github repos. A
separate to-do to fix in the process.

**Comment by Garbee** [comment]
I just tried the "devtools.chrome.com" shortcut. It is now redirecting to the HTTPS version if you try HTTP and the HTTPS version isn't forwarding to the actual docs.

Was HSTS turned on recently? I'm noticing that is the source of the redirect in my network panel.
